src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInMetricsListener.cs

```diff
@@ -81,6 +99,22 @@ public override void OnEventWritten(string name, object payload)
                     tags.Add(new KeyValuePair<string, object>(SemanticConventions.AttributeHttpRoute, route));
                 }
 #endif
+                if (this.options.Enrich != null)
+                {
+                    try
+                    {
+                        this.options.Enrich(context, out var enrichedTags);
```

Comment:
@alanwest @utpilla One question. The current design we call `Enrich` which returns tags (using `out` parameter). The returned tags are added to the list curated by the instrumentation. This is subtly different from the `Activity` enrich callbacks where we pass the `Activity` allowing for full modification of tags added by the instrumentation. Perhaps instead of `out` we should pass the `tags` above as a `ref` to the callback which would allow for addition & modification akin to tracing?

Reply:
> Perhaps instead of out we should pass the tags above as a ref to the callback which would allow for addition & modification akin to tracing?

I think that's a good idea.

Reply:
Reworked as ref